shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -154,18 +156,31 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
 
 - (void)showShareViewController:(NSString*)content {
   UIViewController* engineViewController = [_engine.get() viewController];
+
   NSArray* itemsToShare = @[ content ?: [NSNull null] ];
   UIActivityViewController* activityViewController =
       [[[UIActivityViewController alloc] initWithActivityItems:itemsToShare
                                          applicationActivities:nil] autorelease];
 
   if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad) {
+    // On iPad, the share screen is presented in a popover view, and requires a CGRect
+    FlutterTextInputPlugin* _textInputPlugin = [_engine.get() textInputPlugin];
+    UITextRange* range = _textInputPlugin.textInputView.selectedTextRange;
+
+    CGRect firstRect = [(FlutterTextInputView*)_textInputPlugin.textInputView
+        caretRectForPosition:(FlutterTextPosition*)range.start];
```

Comment:
take a look at `firstRectForRange` which takes the range directly.

Reply:
It doesnt, it only gives me the rect for the first character. The size is also off

Reply:
can you add a note here that we can't use `firstRectForRange` API since it's current implementation doesn't always return the full rect of the range.